modules/data-streams/src/main/java/org/elasticsearch/datastreams/lifecycle/downsampling/DeleteSourceAndAddDownsampleToDS.java

```diff
@@ -72,9 +76,9 @@ ClusterState execute(ClusterState state) {
             );
             return state;
         }
-        IndexMetadata sourceIndexMeta = state.metadata().index(sourceBackingIndex);
-        DataStream dataStream = state.metadata().dataStreams().get(dataStreamName);
+        IndexAbstraction sourceIndexAbstraction = state.metadata().getIndicesLookup().get(sourceBackingIndex);
         if (sourceIndexAbstraction == null) {
```

Comment:
Now that this is atomic, can we still get here? Should this instead just be an assertion b/c we've done something wrong if we wind up here?

Reply:
We can still get here because by the time we get to execute the task, a user might've manually deleted the index. This is still a valid branch.